client/benches/torii.rs

```diff
@@ -52,7 +41,10 @@ fn query_requests(criterion: &mut Criterion) {
             )
             .build(),
         &chain_id,
-        &get_genesis_key_pair(&configuration),
+        configuration
+            .genesis
+            .key_pair()
```

Comment:
as I mentioned previously, sometimes we have getters and sometimes we don't. Also the problem is that config fields are interdependent. If we used getters and setters we can make sure that all invariants are upheld when modifying the configuration

To sum up, I would prefer config to have private fields and getters for the following reasons:
1. consistency - most of our public API is accessed through getters
2. maintaining invariants
3. future proofing

setters we can but don't even have to have. I wouldn't mind if the config structs were completely immutable

Reply:
`key_pair()` returns an `Option<&KeyPair>`, as I mentioned before. It is a useful way to universally access the key pair from a genesis config which might be either `Partial` or `Full`. It is a convenience, same as computing `peer_id()` from actual `address` and `public_key`, or computing `pipeline_time` from `block_time` and `commit_time`.

I don't know how else to reply. I am confused by using these getters/setters conventions in this case...